packages/aztec.js/src/proof/burn/verifier.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Verification algorithm for AZTEC burn zero-knowledge proofs
+ *
+ * @namespace verifier
+ * @memberof module:proof.burn
+ */
+const utils = require('@aztec/dev-utils');
+const BN = require('bn.js');
+const { padLeft } = require('web3-utils');
+
+const bn128 = require('../../bn128');
+const Keccak = require('../../keccak');
+const proofUtils = require('../proofUtils');
+
+const { ERROR_TYPES } = utils.constants;
+
+const { groupReduction } = bn128;
+
+const verifier = {};
+
+/**
+ * Verify an AZTEC burn zero-knowledge proof
+ *
+ * @method verifyProof
+ * @memberof module:proof.burn.verifier
+ * @param {string[]} proofData AZTEC join-split zero-knowledge proof data
+ * @param {number} m number of input notes
+ * @param {string} challengeHex hex-string formatted proof challenge
+ * @param {string} sender Ethereum address of transaction sender
+ */
+verifier.verifyProof = (proofData, challengeHex, sender) => {
+    const m = 1;
+    const errors = [];
+
+    const {
+        rollingHash,
+        notes,
+        challenge,
+    } = proofUtils.convertTranscript(proofData, m, challengeHex, errors, 'burn');
+
+    const finalHash = new Keccak();
```

Comment:
see above, this should validate that m = 1 and n >= 2, then pass the proof to joinSplit/verifier to verify

Reply:
Done